src/Parser/TypeParser.php

```diff
@@ -476,7 +475,65 @@ private function parseCallable(TokenIterator $tokens, Ast\Type\IdentifierTypeNod
 		$startIndex = $tokens->currentTokenIndex();
 		$returnType = $this->enrichWithAttributes($tokens, $this->parseCallableReturnType($tokens), $startLine, $startIndex);
 
-		return new Ast\Type\CallableTypeNode($identifier, $parameters, $returnType);
+		return new Ast\Type\CallableTypeNode($identifier, $parameters, $returnType, $templates);
+	}
+
+
+	/**
+	 * @return Ast\Type\CallableTypeTemplateNode[]
+	 *
+	 * @phpstan-impure
+	 */
+	private function parseCallableTemplates(TokenIterator $tokens): array
```

Comment:
I'm wondering if this should be tied to `CallableType`? I believe the concept is more universal, and could later be reused in other places where an inline declaration of template types might be useful, such as in generic type aliases 🤔

Reply:
So you prefer to call it "template" only? @ondrejmirtes should decide, as it might be confusing with "generic" later...

Can you post any phpdoc example, outside callable, where a template can make sense?

Reply:
Haven't thought about the naming, honestly. As for the example, as I've said, generic type aliases could be a candidate:

```php
/**
 * @phpstan-type Factory<T> = callable(): T
 */
```

I'm not planning to implement them anytime soon (if at all), but it shows that the concept of inline template type declarations could be reused. That being said, it can easily be refactored when/if the need arises